### PR TITLE
entities: update-label: allow to delete a label

### PR DESCRIPTION
**Pull request by maxlath**
This is currently not possible, which can be an issue in cases where a label was set in a language where there is no label for a given work. Example: [this work](https://inventaire.io/api/entities?action=by-uris&uris=inv:320d4c7c453886081a4da182e3d1e512) has a label in Catalan but no know label in Spanish, thus the attempt by a contributor to delete the Spanish label.

### CODE REVIEW

**Comment by jum-s** [comment]
May i question why coming with a API change instead of a new action ?

Even though, this  updating to null = delete seems to exist elsewhere in the app, it explicitly are some explicit guard (like in [group model](https://github.com/inventaire/inventaire/blob/master/server/models/attributes/group.js#L20)).
A use case i see problematic is that forgetting to populate the new value in a request ends up in an unfortunate update which remove some data.
I kind of liked/got used to the explicit ways, like `updateClaim` with its `old-value`/`new-value` mechanism

Why not opening an explicit new action `deleteLabel` instead?

**Comment by maxlath** [comment]
> forgetting to populate the new value in a request ends up in an unfortunate update which remove some data

no, because `null !== undefined`: not populating `value` would make it undefined, which is still rejected; it would have to be explicitly set to `null` to take effect

**Comment by jum-s** [comment]
red test
```
  1) entities:update-labels
       should reject an up-to-date value:

      AssertionError: expected 429 to be 400
      + expected - actual

      -429
      +400
```

**Comment by jum-s** [comment]
> it would have to be explicitly set to `null` to take effect

still more error prone than an explicit split . A topic that was interestingly discussed #98 some time ago.

**Comment by jum-s** [comment]
i dont get why a [`canBeNull` option](https://github.com/inventaire/inventaire/blob/master/server/controllers/groups/update_settings.js#L9) is not necessary in the `updateLabel` controller. I guess it would document a bit that null is possible action

**Comment by maxlath** [comment]
> i dont get why a [canBeNull option](https://github.com/inventaire/inventaire/blob/master/server/controllers/groups/update_settings.js#L9) is not necessary

because `optional=true`, but indeed `canBeNull` would be a better fit if we stay with that implementation

**Comment by maxlath** [comment]
I tried to address the concern on the API in e55470631

File: server/models/entity.js [hunk 1]
```diff
@@ -284,3 +288,15 @@ const setPossiblyEmptyPropertyArray = (doc, property, propertyArray) => {
     doc.claims[property] = propertyArray
   }
 }
+
+const deleteLabel = (doc, lang) => {
+  if (doc.labels[lang] == null) {
+    throw error_.new('can not delete a non-existant label', 400, { doc, lang })
```
Comment (jum-s) [inline server/models/entity.js:294]:
```suggestion
    throw error_.new('can not delete a label which do not exist', 400, { doc, lang })
```

File: server/models/entity.js [hunk 2]
```diff
@@ -284,3 +288,15 @@ const setPossiblyEmptyPropertyArray = (doc, property, propertyArray) => {
...
+  }
+
+  if (Object.keys(doc.labels).length === 1) {
+    throw error_.new('can not delete the last label', 400, { doc, lang })
```
Comment (jum-s) [inline server/models/entity.js:298]:
```suggestion
    throw error_.new('can not delete the last remaining label', 400, { doc, lang })
```

**Comment by jum-s** [comment]
> I tried to address the concern on the API in [e554706](https://github.com/inventaire/inventaire/commit/e55470631d2b767e67affb8db7b8e96c29b48ed4)

This commit doesnt belong to this PR, is that on purpose ?

**Comment by maxlath** [comment]
@jum-s not sure what happened then, but after rebase, e554706 is now 6b2aa85 and is part of this PR

**Review by jum-s — APPROVED** [review]

**Pull request merged**